### DISCUSSION TITLE
UI guidelines for an extension that requires configuration

### DISCUSSION THREAD

**Discussion by skybrian** [microsoft/vscode-discussions#756] [Extension Development QnA] [answered]
I have an extension that depends on a command-line utility being installed. Most users probably don't have it. Once they do have it installed, they need to set its path in a configuration property. This means that the extension might start out in an unconfigured state. My extension can detect this.

What's the best approach for informing the user about the situation and how to fix it? It's not urgent (my extension isn't that important), but they won't be able to use the extension without it. I'd like to make the current state obvious so they don't think it's broken, without being obtrusive. Is this an info notification? A status bar item?

I guess if they try to run a command, it could show an error notification. There's also an autocomplete. It'd be a bit odd to offer to do the autocompletion and error out.

**Comment by isidorn** [comment, marked as answer]
I think the notification is the way to go here.
Though try to show the notification at the precisest moment possible. So definitely not on startup, but lazily when you actually require that dependency (and it is ideally triggered by a user action)?

If your extension already uses smart lazy activation events, then you might do this check in your extension activates to get the desired effect.

Hope this helps
